File: src/main/java/nl/crosscode/x509certbuddy/ui/CBActionManager.java
package nl.crosscode.x509certbuddy.ui;

import com.intellij.icons.AllIcons;
import com.intellij.openapi.actionSystem.ActionGroup;
import com.intellij.openapi.actionSystem.DefaultActionGroup;
import nl.crosscode.x509certbuddy.actions.*;

public class CBActionManager {


    static synchronized ActionGroup buildContextMenu(Exporters exporters) {

        DefaultActionGroup exportActions = new DefaultActionGroup();
        exportActions.setPopup(true);
        exportActions.getTemplatePresentation().setText("Export");
        exportActions.getTemplatePresentation().setIcon(AllIcons.Actions.Download);
        exportActions.add(new ExportPEMAction(exporters));
        exportActions.add(new ExportCertChainPEMAction(exporters));
        exportActions.add(new ExportDERAction(exporters));
        exportActions.add(new ExportAllAction(exporters));

        DefaultActionGroup copyActions = new DefaultActionGroup();
        copyActions.setPopup(true);
        copyActions.getTemplatePresentation().setText("Copy");
        copyActions.getTemplatePresentation().setIcon(AllIcons.Actions.Copy);
        copyActions.add(new CopyPEMAction(exporters));
        copyActions.add(new CopyCertChainPEMAction(exporters));
        copyActions.add(new CopyBase64Action(exporters));
        copyActions.add(new CopyAllAction(exporters));

        DefaultActionGroup treeActions = new DefaultActionGroup();
        treeActions.add(copyActions);
        treeActions.add(exportActions);
        treeActions.add(new RemoveSelectedCertAction());
        treeActions.add(new RemoveAllCertsAction());
        return treeActions;
    }
}

File: src/main/kotlin/nl/crosscode/x509certbuddy/ui/CBActionManager.kt
package nl.crosscode.x509certbuddy.ui

import com.intellij.icons.AllIcons
import com.intellij.openapi.actionSystem.ActionGroup
import com.intellij.openapi.actionSystem.DefaultActionGroup
import nl.crosscode.x509certbuddy.actions.*

object CBActionManager {
    @JvmStatic
    @Synchronized
    fun buildContextMenu(exporters: Exporters?): ActionGroup {
        val exportActions = DefaultActionGroup()
        exportActions.isPopup = true
        exportActions.templatePresentation.text = "Export"
        exportActions.templatePresentation.icon = AllIcons.Actions.Download
        exportActions.add(ExportPEMAction(exporters!!))
        exportActions.add(ExportCertChainPEMAction(exporters))
        exportActions.add(ExportDERAction(exporters))
        exportActions.add(ExportAllAction(exporters))

        val copyActions = DefaultActionGroup()
        copyActions.isPopup = true
        copyActions.templatePresentation.text = "Copy"
        copyActions.templatePresentation.icon = AllIcons.Actions.Copy
        copyActions.add(CopyPEMAction(exporters))
        copyActions.add(CopyCertChainPEMAction(exporters))
        copyActions.add(CopyBase64Action(exporters))
        copyActions.add(CopyAllAction(exporters))

        val treeActions = DefaultActionGroup()
        treeActions.add(copyActions)
        treeActions.add(exportActions)
        treeActions.add(RemoveSelectedCertAction())
        treeActions.add(RemoveAllCertsAction())
        return treeActions
    }
}
